src/tests/Interop/Swift/SwiftInvalidCallConv/SwiftInvalidCallConv.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Swift;
+using Xunit;
+
+public class InvalidCallingConvTests
+{
+    private const string SwiftLib = "libSwiftInvalidCallConv.dylib";
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(SwiftLib, EntryPoint = "$s20SwiftInvalidCallConv10simpleFuncyyF")]
+    public static extern nint FuncWithTwoSelfParameters(SwiftSelf self1, SwiftSelf self2);
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(SwiftLib, EntryPoint = "$s20SwiftInvalidCallConv10simpleFuncyyF")]
+    public unsafe static extern nint FuncWithTwoErrorParameters(SwiftError* error1, SwiftError* error2);
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(SwiftLib, EntryPoint = "$s20SwiftInvalidCallConv10simpleFuncyyF")]
+    public unsafe static extern nint FuncWithMixedParameters(SwiftSelf self1, SwiftSelf self2, SwiftError* error1, SwiftError* error2);
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(SwiftLib, EntryPoint = "$s20SwiftInvalidCallConv10simpleFuncyyF")]
+    public static extern nint FuncWithSwiftErrorAsArg(SwiftError error1);
+
+    [Fact]
+    public static void TestFuncWithTwoSelfParameters()
+    {
+        SwiftSelf self = new SwiftSelf(IntPtr.Zero);
+
+        try
+        {
+            FuncWithTwoSelfParameters(self, self);
+            Assert.Fail("FuncWithTwoSelfParameters should have thrown InvalidProgramException. Invalid due to multiple SwiftSelf arguments.");
+        }
+        catch (InvalidProgramException e) { }
```

Comment:
Use `Assert.Throws` instead of a try-catch?
```suggestion
        Assert.Throws<InvalidProgramException>(() => FuncWithTwoSelfParameters(self, self));
```